Slim/Player/Protocols/HTTP.pm

```diff
@@ -328,9 +328,12 @@ sub parseDirectHeaders {
 	my ( $self, $client, $url, @headers ) = @_;
 
 	my $isDebug = main::DEBUGLOG && $directlog->is_debug;
-
+	my $oggType;
+	
 	# May get a track object
 	if ( blessed($url) ) {
+		($oggType) = $url->content_type =~ /(ogf|ogg|ops)/;
```

Comment:
Trying to understand what you're doing here: what would `$url->content_type` be here? Couldn't we just set `$contentType` here, and only set it from the header if it wasn't defined yet? `$oggType` would take precedence over `$contentType` anyway.

Reply:
It's because $url->content_type is defined by Remote::parseOggHeader by using the real type of the stream, not by just the HTTP header content_type/mimetype. So when we have a blessed track and if the content_type is set, it's because we have the proper format acquired by streaming some bytes.

Now, if we set contentType there, then in line ~431 of HTTP.pm, 
```$contentType = Slim::Music::Info::mimeToType($contentType);```

We will fail because here the expected parameter is a real mimeType, not just the 3 letters to the format (it uses types.conf as a hash to build matching from mimetypes to 3-letters type). 

So there, I need the encapsulation format, which is audio/ogg or application/ogg. But we want parseDirectHeaders to return 3 letters type only (not the mimetype) and we want it to return the *real* 3 letters, no the container letters. Maybe a better test on line 335 would be to populate oggType by $url->content_type only if HTTP header's contentType contains "/ogg", Might be even more future proof

Hope it makes sense

Reply:
Yep, makes sense. May I ask you (again) to rebase this change against LMS8? I'll backport it if it proves to be good.